src/hostname/hostnamed.c

```diff
@@ -1241,6 +1299,122 @@ static int validate_and_substitute_hostname(const char *name, char **ret_substit
         return 1;
 }
 
+static bool validate_ansi_color(const char *color) {
+        if (isempty(color))
+                return true;
+
+        if (string_has_cc(color, NULL))
+                return false;
+        _cleanup_strv_free_ char **parts = NULL;
+        int r = strv_split_full(&parts, color, ";", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return false;
+
+        if (strv_length(parts) < 2)
+                return false;
+
+        unsigned code;
+        r = safe_atou(parts[0], &code);
+        if (r < 0)
+                return false;
+
+        if (code == 0) {
+                if (strv_length(parts) != 2)
+                        return false;
+                r = safe_atou(parts[1], &code);
+                if (r < 0 || code > 255)
+                        return false;
+                return true;
+        }
+
+        if (code == 38 || code == 48) {
+                r = safe_atou(parts[1], &code);
+                if (r < 0 || (code != 2 && code != 5))
+                        return false;
+
+                if (code == 2) {
+                        if (strv_length(parts) != 5)
+                                return false;
+                        for (int i = 2; i < 5; i++) {
+                                r = safe_atou(parts[i], &code);
+                                if (r < 0 || code > 255)
+                                        return false;
+                        }
+                        return true;
+                }
+
+                if (code == 5) {
+                        if (strv_length(parts) != 3)
+                                return false;
+                        r = safe_atou(parts[2], &code);
+                        if (r < 0 || code > 255)
+                                return false;
+                        return true;
+                }
+        }
+
+        return false;
+}
+
+static int method_set_ansi_color(sd_bus_message *m,void *userdata, sd_bus_error *error) {
+        Context *c = ASSERT_PTR(userdata);
+        const char *color;
+        int interactive, r;
+
+        assert(m);
+
+        r = sd_bus_message_read(m, "sb", &color, &interactive);
```

Comment:
you dropped the "interactive" param from the method in the introspection data below. please drop it here too